client/utils/moving_statistics/moving_average.c

```diff
@@ -13,13 +13,14 @@ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
 #include "moving_average.h"
 
-void welford_one_pass(long double*, long double, uint16_t);
-
 // this function update the (old) mean, adding a new sample to this mean.
 // it recalculate the mean without keep an array of all old samples
-void welford_one_pass(long double* mean, long double sample, uint16_t nsamples){
-	if(nsamples)
-        	*mean=*mean+(sample-*mean)/(long double)(nsamples);
+static long double welford_one_pass(long double mean, long double sample, uint16_t nsamples)
```

Comment:
It makes sense to return result through parameter only if function already returns something. Otherwise it is better and more clear to do standard "return"